contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -3073,6 +3073,134 @@ CREATE OR REPLACE FUNCTION sys.substring(string sys.NCHAR, i INTEGER, j INTEGER)
 RETURNS sys.NVARCHAR
 AS 'babelfishpg_tsql', 'tsql_varchar_substr' LANGUAGE C IMMUTABLE PARALLEL SAFE;
 
+-- wrapper functions for upper --
+-- Function to handle datatypes which are implicitly convertable to VARCHAR
+CREATE OR REPLACE FUNCTION sys.upper(ANYELEMENT)
+RETURNS sys.VARCHAR
+AS $$
+DECLARE
+    type_oid oid;
+    typ_base_oid oid;
+    typnam text;
+BEGIN
+    typnam := NULL;
+    type_oid := pg_typeof($1);
+    typnam := sys.translate_pg_type_to_tsql(type_oid);
+    IF typnam IS NULL THEN
+        typ_base_oid := sys.bbf_get_immediate_base_type_of_UDT(type_oid);
+        typnam := sys.translate_pg_type_to_tsql(typ_base_oid);
+    END IF;
```

Comment:
Have we validated that UDT should be allowed as per TSQL?

Reply:
Yes we have validated this with TSQL